bashbrew/go/src/bashbrew/cmd-put-shared.go

```diff
@@ -21,24 +60,55 @@ func cmdPutShared(c *cli.Context) error {
 		return fmt.Errorf(`"--namespace" is a required flag for "put-shared"`)
 	}
 
-	fmt.Fprintf(os.Stderr, "warning: this subcommand is still a big WIP -- it doesn't do anything yet!\n")
-
 	for _, repo := range repos {
 		r, err := fetch(repo)
 		if err != nil {
 			return cli.NewMultiError(fmt.Errorf(`failed fetching repo %q`, repo), err)
 		}
 
-		// TODO handle all multi-architecture tags first (regardless of whether they have SharedTags)
+		// handle all multi-architecture tags first (regardless of whether they have SharedTags)
+		for _, entry := range r.Entries() {
+			// "image:" will be added later so we don't have to regenerate the entire "manifests" section every time
+			yaml, err := entriesToManifestToolYaml(*r, &entry)
+			if err != nil {
+				return err
+			}
+
+			for _, tag := range r.Tags(namespace, false, entry) {
+				tagYaml := fmt.Sprintf("image: %s\n%s", tag, yaml)
+				fmt.Printf("Putting %s\n", tag)
+				if err := manifestToolPushFromSpec(tagYaml); err != nil {
+					return fmt.Errorf("failed pushing %q (%q)", tag, entry.TagsString())
+				}
+			}
```

Comment:
See https://github.com/estesp/manifest-tool/issues/23 for some discussion of making `manifest-tool` more efficient for our use case (and thus removing the need for so many nested loops :innocent:).